### PR TITLE
Fix: Register Filter Blocks using `block.json`

### LINKED ISSUE #6491: Register Filter Blocks using `block.json`
For each block:
- Add `block.json` file for each filter block.
- Register JS block using metadata from `block.json`.

Blocks to be updated:

- [ ] Active Filters
- [ ] Filter Products by Prices
- [ ] Filter Products by Attribute
- [ ] Filter Products by Stock

Part of https://github.com/woocommerce/woocommerce-blocks/issues/4672

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6491

This PR:
- Scans and copies the block.json file automatically to the build folder instead of hard-coding the file list (cc @senadir because you recently edited those entries).
- Create `block.json` and register the block using metadata for filter blocks.

Note:
- Translation for block title and description is handled by `po` file, not JSON.
- `make-pot` doesn't pick the default value of attributes, so we have to override them on the JS side.

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) .
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md)
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

N/a

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Check out and build this branch.
2. Edit a page, and see all filter blocks are working.
3. See those blocks work on the front end too.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Register filter blocks using block metadata.

### CODE REVIEW

File: bin/webpack-configs.js [hunk 1]
```diff
@@ -260,32 +260,16 @@ const getMainConfig = ( options = {} ) => {
...
+						from: './assets/js/blocks/**/block.json',
+						to( { absoluteFilename } ) {
+							const blockName = absoluteFilename
+								.split( '/' )
+								.at( -2 );
+							return `./${ blockName }/block.json`;
+						},
+						globOptions: {
+							ignore: [ '**/inner-blocks/**' ],
+						},
```
Comment (senadir) [inline bin/webpack-configs.js:272]: To have a cleaner (and less conflict) folder structure, we might want to pivot away from `${ blockName }/block.json` toward `${ blockName }.block.json` instead, but I'm not sure how would that play into translation.

> `ignore: [ '**/inner-blocks/**' ],`

I believe lack of having `assets` in our releases (not confirmed?) and not having them in `build` is the reason why block metadata translation isn't loading in the editor :/

Comment (dinhtungdu) [reply bin/webpack-configs.js:272]: > I believe lack of having assets in our releases (not confirmed?) and not having them in build is the reason why block metadata translation isn't loading in the editor :/

We only need `block.json` files in the `build` to translate the block metadata. `asset` is unnecessary to get translation working in my test.

> we might want to pivot away from ${ blockName }/block.json toward ${ blockName }.block.json instead, but I'm not sure how would that play into translation.

I will test this. But it's possible that `make-pot` scans only `block.json` files.

Comment (dinhtungdu) [reply bin/webpack-configs.js:272]: > I will test this. But it's possible that `make-pot` scans only `block.json` files.

It is. See https://github.com/wp-cli/i18n-command/blob/main/src/MakePotCommand.php#L666-L679

**Review by Aljullu — COMMENTED** [review]
Thanks for working on this, @dinhtungdu! I have a couple of questions:

> Translation for block title and description is handled by `po` file, not JSON.

That means we will break existing translations, right? At least until this version of WC Blocks is merged into WC core and translations are generated again?

> `make-pot` doesn't pick the default value of attributes, so we have to override them on the JS side.

Just curious, couldn't we just declare those attributes in the JS file, then? Or is there any benefit on duplicating them between the JSON and JS files?

**Comment by dinhtungdu** [comment]
> > Translation for block title and description is handled by `po` file, not JSON.
> 
> That means we will break existing translations, right? At least until this version of WC Blocks is merged into WC core and translations are generated again?

@Aljullu Yes.

> Just curious, couldn't we just declare those attributes in the JS file, then? Or is there any benefit on duplicating them between the JSON and JS files?

Yes. I did that so we can grasp all block attributes by looking only at the block.json file. I admit that it also creates confusion at the same time. I don't have a strong opinion on this, I'm happy to delete the duplicated entry in the block.json if it makes more sense.

**Comment by Aljullu** [comment]
> @Aljullu Yes.

Thanks for clarifying! That's unfortunate, do you think we could work around it? One idea would be to:

- Add the strings to `block.json`, as you are doing in this PR.
- For now, keep the strings in `index.js`, so we override them with the localized ones.
- Once the translations have been updated, remove the strings from `index.js` (we will need to create a follow-up issue for that).

Do you think that will work?

> Yes. I did that so we can grasp all block attributes by looking only at the block.json file. I admit that it also creates confusion at the same time. I don't have a strong opinion on this, I'm happy to delete the duplicated entry in the block.json if it makes more sense.

I don't have a strong opinion either. :smile:  I think your solution makes sense, but I might remove the `default` value from the JSON files so it's only defined in the JS ones. The reason is that this way we have the strings defined only in one place. But as I said, I don't have a strong opinion on this, so I'm fine with any decision.

**Comment by dinhtungdu** [comment]
@Aljullu I address the translation issue per your suggestion. Thanks for that! After consideration, I also remove the duplicated attribute entries because they create more confusion than they solve.

**Review by Aljullu — APPROVED** [review]
LGTM, thanks for updating the PR, @dinhtungdu!

> - Once the translations have been updated, remove the strings from `index.js` (we will need to create a follow-up issue for that).

Will you be able to create an issue for this one so we don't forget? :point_up:

**Comment by dinhtungdu** [comment]
> Will you be able to create an issue for this one so we don't forget? ☝️
 
I created #6543 for this.

**Pull request merged**